src/server/lib/__tests__/getABForcedVariants.test.ts

```diff
@@ -1,19 +1,15 @@
 import { Request } from 'express';
 import { getABForcedVariants } from '@/server/lib/getABForcedVariants';
 
-const fakeRequest = (queryParameters?: string[]) => ({
-  url: `https://guardian.com${
-    queryParameters && '?' + queryParameters.join('&')
-  }`,
-});
-
 describe('getABForcedVariants', () => {
   describe('when there are variants', () => {
     it('returns the variants stripped of their ab- prefix', () => {
-      const request = fakeRequest([
-        'ab-test1=testparameter1',
-        'ab-test2=testparameter2',
-      ]) as unknown as Request;
+      const request = {
+        query: {
+          'ab-test1': 'testparameter1',
+          'ab-test2': 'testparameter2',
+        },
+      } as unknown as Request;
```

Comment:
How does `as unknown as Request` work here?

Reply:
It's basically a bit of a hack to avoid having to generate the whole `Request` object.

It's first saying parse this object as the `unknown` type which is a catch all for all valid types, then from the `unknown` type parse it as anything else, so in this case the `Request` type.

https://basarat.gitbook.io/typescript/type-system/type-assertion#double-assertion

Reply:
As this is for a test, this is fine, but wouldn't really want to use this in actual code!

Reply:
Very useful trick though. 👍